batch/batch/server/server.py

```diff
@@ -791,35 +766,45 @@ def from_record(record):
                                        ttl=ttl,
                                        is_open=is_open,
                                        userdata=json.dumps(userdata),
-                                       user=user)
+                                       user=user,
+                                       deleted=False)
 
         batch = Batch(id=id, attributes=attributes, callback=callback,
-                      ttl=ttl, is_open=is_open, userdata=userdata, user=user)
+                      ttl=ttl, is_open=is_open, userdata=userdata, user=user,
+                      state='running', complete=False, deleted=False)
         batch.close_ttl()
         return batch
 
-    def __init__(self, id, attributes, callback, ttl, is_open, userdata, user):
+    def __init__(self, id, attributes, callback, ttl, is_open, userdata, user,
+                 state, complete, deleted):
         self.id = id
         self.attributes = attributes
         self.callback = callback
         self.ttl = ttl
         self.is_open = is_open
         self.userdata = userdata
         self.user = user
+        self.state = state
+        self.complete = complete
+        self.deleted = deleted
 
     async def get_jobs(self):
         return [Job.from_record(record) for record in await db.jobs.get_records_by_batch(self.id)]
 
     async def cancel(self):
+        await self.close()
         jobs = await self.get_jobs()
         for j in jobs:
             await j.cancel()
 
     async def delete(self):
+        # Batch is deleted from db in polling loop once all jobs are complete
+        await db.batch.update_record(self.id,
+                                     deleted=True,
+                                     is_open=False)
         jobs = await self.get_jobs()
         for j in jobs:
-            assert j.batch_id == self.id
-            await db.jobs.update_record(j.id, batch_id=None)
+            await j.delete()
```

Comment:
I think we can remove Job.delete and just cancel here.  The delete cleanup logic will apply after everything winds down.

Reply:
I think this should be a call to cancel, see above comment.

Reply:
The difference between delete and cancel is the log files are deleted in `delete` but not in `cancel`.